control/control.go

```diff
@@ -723,94 +831,57 @@ func (p *pluginControl) validateMetricTypeSubscription(mt core.RequestedMetric,
 	return serrs
 }
 
-type gatheredPlugin struct {
-	plugin           core.Plugin
-	subscriptionType strategy.SubscriptionType
-}
-
-func (p *pluginControl) gatherCollectors(mts []core.Metric) ([]gatheredPlugin, []serror.SnapError) {
-	var (
-		plugins []gatheredPlugin
-		serrs   []serror.SnapError
-	)
-	// here we resolve and retrieve plugins for each metric type.
-	// if the incoming metric type version is < 1, we treat that as
-	// latest as with plugins.  The following two loops create a set
-	// of plugins with proper versions needed to discern the subscription
-	// types.
-	colPlugins := make(map[string]gatheredPlugin)
-	for _, mt := range mts {
-		// If the version provided is <1 we will get the latest
-		// plugin for the given metric.
-		m, err := p.metricCatalog.Get(mt.Namespace(), mt.Version())
-		if err != nil {
-			serrs = append(serrs, serror.New(err, map[string]interface{}{
-				"name":    mt.Namespace().String(),
-				"version": mt.Version(),
-			}))
-			continue
-		}
-		subType := strategy.BoundSubscriptionType
-		if mt.Version() < 1 {
-			subType = strategy.UnboundSubscriptionType
-		}
-		colPlugins[fmt.Sprintf("%s:%d", m.Plugin.Key(), subType)] = gatheredPlugin{
-			plugin:           m.Plugin,
-			subscriptionType: subType,
-		}
-	}
-	if len(serrs) > 0 {
-		return plugins, serrs
-	}
-
-	for _, lp := range colPlugins {
-		plugins = append(plugins, lp)
-	}
-	if len(plugins) == 0 {
-		serrs = append(serrs, serror.New(errors.New("No plugins found")))
-		return nil, serrs
-	}
-	return plugins, nil
-}
-
-func (p *pluginControl) SubscribeDeps(taskID string, mts []core.Metric, plugins []core.Plugin) []serror.SnapError {
+func (p *pluginControl) SubscribeDeps(taskID string, mts []core.Metric, requested []core.RequestedMetric, plugins []core.Plugin, configTree *cdata.ConfigDataTree) []serror.SnapError {
```

Comment:
Why core.Metric and requested.Metric, when requested.Metric is a subset of core.Metric? From the call-ees point of view what is the difference when using one vs the other?